r/R/dplyr.R

```diff
@@ -73,6 +80,22 @@ print.arrow_dplyr_query <- function(x, ...) {
   invisible(x)
 }
 
+get_field_names <- function(selected_cols) {
+  if (inherits(selected_cols, "arrow_dplyr_query")) {
+    selected_cols <- selected_cols$selected_columns
+  }
+  map_chr(selected_cols, ~.$field_name %||% .$args$field_name %||% "")
```

Comment:
I can comment, though FWIW I'm pretty sure this will crash once dataset expressions might not be field_refs (in the next PR) so it's likely to be rewritten.

Reply:
See [my other comment](https://github.com/apache/arrow/pull/9521/files#r581514795) in which an error happens on this line

Reply:
I revised this function in https://github.com/apache/arrow/pull/9521/commits/08f78fb41554b2a0095e73a7ee216d6192f77119, PTAL